### PR TITLE
Added .NET SDK for Microsoft.HybridContainerService

### PR DESCRIPTION
**Pull request by HariPrasadV**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/hybridaks/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
+
+trigger:
```
Comment (ArthurMa1978) [inline sdk/hybridaks/ci.yaml:3]: remove this file

File: sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+
+[assembly: Azure.Core.AzureResourceProviderNamespace("HybridContainerService")]
```
Comment (ArthurMa1978) [inline sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Properties/AssemblyInfo.cs:6]: Microsoft.HybridContainerService

File: sdk/hybridaks/ci.yaml (hunk 1 above)
Comment (HariPrasadV) [reply sdk/hybridaks/ci.yaml]: Done!

File: sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (HariPrasadV) [reply sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Properties/AssemblyInfo.cs]: Corrected it now

**Comment by fengzhou-msft** [comment]
Please generate the API file with the command `.\azure-sdk-for-net\eng\scripts\Export-API.ps1 hybridaks`

File: sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/AgentPoolResource.cs [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+
+namespace Azure.ResourceManager.HybridContainerService
+{
+    /// <summary>
+    /// A Class representing an AgentPool along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct an <see cref="AgentPoolResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetAgentPoolResource method.
+    /// Otherwise you can get one from its parent resource <see cref="ProvisionedClustersResponseResource" /> using the GetAgentPool method.
+    /// </summary>
+    public partial class AgentPoolResource : ArmResource
```
Comment (fengzhou-msft) [inline sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/AgentPoolResource.cs:26]: `AgentPool` is a common resource name. Please add a prefix to make it more descriptive. Either use the `prepend-rp-prefix` if `HybridContainerServiceAgentPool` is a good name to use or use `rename-mapping` to use another name in  `autorest.md`. You can check rename configuration examples in [ContainerService autorest.md](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/containerservice/Azure.ResourceManager.ContainerService/src/autorest.md)

File: sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/VirtualNetworkResource.cs [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+
+namespace Azure.ResourceManager.HybridContainerService
+{
+    /// <summary>
+    /// A Class representing a VirtualNetwork along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="VirtualNetworkResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetVirtualNetworkResource method.
+    /// Otherwise you can get one from its parent resource <see cref="ResourceGroupResource" /> using the GetVirtualNetwork method.
+    /// </summary>
+    public partial class VirtualNetworkResource : ArmResource
```
Comment (fengzhou-msft) [inline sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/VirtualNetworkResource.cs:28]: `VirtualNetwork` is too generic. Please add a prefix to the resource name to make it more descriptive.

File: sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/ProvisionedClustersResponseResource.cs [hunk 5]
```diff
@@ -0,0 +1,588 @@
...
+
+namespace Azure.ResourceManager.HybridContainerService
+{
+    /// <summary>
+    /// A Class representing a ProvisionedClustersResponse along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="ProvisionedClustersResponseResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetProvisionedClustersResponseResource method.
+    /// Otherwise you can get one from its parent resource <see cref="ResourceGroupResource" /> using the GetProvisionedClustersResponse method.
+    /// </summary>
+    public partial class ProvisionedClustersResponseResource : ArmResource
```
Comment (fengzhou-msft) [inline sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/ProvisionedClustersResponseResource.cs:28]: The generator takes the response model in swagger as the resource name, but we should get rid of the `Response` suffix as a resource anme. Rename `ProvisionedClustersResponse` to `ProvisionedCluster`. If there is an existing `ProvisionedCluster(s)`, rename that model to something else.

**Comment by ArthurMa1978** [comment]
@HariPrasadV can you update the pr for the comments?

File: sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/AgentPoolResource.cs (hunk 3 above)
Comment (HariPrasadV) [reply sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/AgentPoolResource.cs]: Thanks for the suggestion, I've renamed it now

File: sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/ProvisionedClustersResponseResource.cs (hunk 5 above)
Comment (HariPrasadV) [reply sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/ProvisionedClustersResponseResource.cs]: Thanks for the suggestion, I've renamed it now

File: sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/VirtualNetworkResource.cs (hunk 4 above)
Comment (HariPrasadV) [reply sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/VirtualNetworkResource.cs]: Thanks for the suggestion, I've renamed it now

**Comment by HariPrasadV** [comment]
> @HariPrasadV can you update the pr for the comments?

Sorry for the delay, I've addressed all the comments now, and also tested the SDK and it looks good from our side. Please take a look and let me know if there is anything else. Thanks!

**Review by ArthurMa1978 — APPROVED** [review]

**Comment by ArthurMa1978** [comment]
Merge this pr first, kick off release once the namespace got [approved](https://github.com/Azure/azure-sdk/issues/5057).

**Pull request merged**